### PR TITLE
Family intervene

### PR DESCRIPTION
**Pull request by Kubasinska**
Added family_intervene and other related methods

### CODE REVIEW

File: mesa/Person.py [hunk 1]
```diff
@@ -188,10 +188,7 @@ def remove_professional(self, forlorn): self.remove_link(self, forlorn, 'profess
 
     def age_between(self, low, high):
         return self.age() >= low and self.age() < high
-    
-    def family(self): # maybe add self?
```
Comment (Kubasinska) [inline mesa/Person.py:192]: Changed with Person.family_link_neighbors(), I created this function and realized afterwards that it already existed!

File: mesa/Employer.py [hunk 2]
```diff
@@ -19,10 +19,11 @@ def __init__(self, m: mesaPROTON_OC):
...
         return "Employer: " + str(self.unique_id)
   
-    def create_job(self,  level:int, worker: Person):
-        newjob = Job(level, self, worker, self.m)
+    def create_job(self,  level: int, worker: Person):
+        newjob = Job(self.m)
+        newjob.level = level
         worker.my_job = newjob
-        self.my_jobs.add(newjob)
+        self.my_jobs.append(newjob)
```
Comment (mariopaolucci) [inline mesa/Employer.py:26]: You really didn't like my `Set`s...

File: mesa/extra.py [hunk 3]
```diff
@@ -49,9 +49,11 @@ def social_proximity(ego:Person, alter:Person):
...
     return acc
 
-def at_most(n, a, rng_istance, replace=True):
-    return a if len(a) < n else rng_istance.choice(a,n,replace=replace)
-
+def at_most(agentset, n, rng_istance):
+    if len(agentset) < n:
+        return agentset
+    else:
+        return list(rng_istance.choice(agentset,n, replace=False))
```
Comment (mariopaolucci) [inline mesa/extra.py:56]: whitespace `,n, repla`

File: mesa/mesaPROTON_OC.py [hunk 4]
```diff
@@ -53,6 +53,11 @@ def __init__(self, seed=None):
         self.labour_status_by_age_and_sex = 0
         self.labour_status_range = 0
 
+        self.removed_fatherships = list()
```
Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:56]: I don't get the whitepspacing here.. removed_fathership should go under some category (I'd keep it in the interventions area)

File: mesa/mesaPROTON_OC.py [hunk 5]
```diff
@@ -246,26 +259,38 @@ def socialization_intervene(self):
...
-        for x in targets:
-            support_set = extra.at_most(50, [y for y in schedule.agents if y.num_crimes_committed == 0 and y.age() > x.age()], self.rng)
-        if support_set:
-            chosen = self.rng.choice(support_set,
-                                      p=[(1 - (y.age() - x.age()) / 120) for y in support_set],
-                                      size=None,
-                                      replace=False)
-            chosen.makeFriends(x)
+        for agent in targets:
+            support_set = extra.at_most([support_agent for support_agent in self.schedule.agents if support_agent.num_crimes_committed == 0 and support_agent.age() > agent.age() and support_agent != agent], 50,  self.rng)
```
Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:278]: double spaces `50,  `

Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:278]: Remove unnecessary `support_agent != agent` (same age)

File: mesa/mesaPROTON_OC.py [hunk 6]
```diff
@@ -286,37 +311,58 @@ def welfare_intervene(self):
...
+            kids_pool = list(self.rng.choice(kids_to_protect, how_many, replace=False))
+            for kid in kids_pool:
+                self.kids_intervention_counter += 1
+                # notice that the intervention acts on ALL family members respecting the condition, causing double calls for families with double targets.
+                # gee but how comes that it increases with the nubmer of targets? We have to do better here
                 # this also removes household links, leaving the household in an incoherent state.
-                x.neighbor.get('parent').remove(x.father)
-                # maybe not needed?
-                self.removed_fatherships.add([((18 * ticks_per_year + birth_tick) - ticks), x.father, x])
+                kid.neighbors.get("parent").remove(kid.father)
```
Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:357]: Not updating the `father` variable is planned? Add comment in case

**Review by mariopaolucci — CHANGES_REQUESTED** [review]

**Comment by Kubasinska** [comment]
Thanks for the comments, changes made, I await confirmation for the merge.

**Review by mariopaolucci — APPROVED** [review]

**Pull request merged**